src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -2,25 +2,54 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Headers;
+using Microsoft.AspNetCore.Internal;
 using Microsoft.Extensions.Logging;
 using Microsoft.Net.Http.Headers;
 
 namespace Microsoft.AspNetCore.Mvc.Internal
 {
     public class FileResultExecutorBase
     {
+        private const string AcceptRangeHeaderValue = "bytes";
+
         public FileResultExecutorBase(ILogger logger)
         {
             Logger = logger;
         }
 
         protected ILogger Logger { get; }
 
-        protected void SetHeadersAndLog(ActionContext context, FileResult result)
+        protected (RangeItemHeaderValue range, long rangeLength)? SetHeadersAndLog(ActionContext context, FileResult result, long fileLength, DateTimeOffset? lastModified = null, EntityTagHeaderValue etag = null)
         {
             SetContentType(context, result);
             SetContentDispositionHeader(context, result);
```

Comment:
I'm not sure if you should send these headers in the cases where you are sending precondition failed or a similar response @Tratcher